awscli/argprocess.py

```diff
@@ -245,7 +245,7 @@ def _is_complex_shape(model):
 
 class ParamShorthand(object):
 
-    def _uses_old_list_case(self, service_name, operation_name, argument_name):
+    def _uses_old_list_case(self, service_id, operation_name, argument_name):
```

Comment:
Could you elaborate on why we need to change shorthand parsing logic to use serviceId's? Looking at the code, `ParamShorthand` is only registered to the [process-cli-arg event](https://github.com/aws/aws-cli/blob/ee71e8cca0838a1505992ca9cd29add33386dfe6/awscli/handlers.py#L90) and that event only ever emits a [service_name](https://github.com/aws/aws-cli/blob/develop/awscli/arguments.py#L461). So I do not think we need to update this.

Reply:
Because we alias both emit and register, so the events are still getting aliased. These in particular have to be changed because they're inspecting the event name. I could also refactor it to be explicit by having the event actually send the service id and operation name as kwargs.